### ISSUE TITLE
Add Repository Contributors

### ISSUE THREAD

**Issue by SysVoid** [serghey-rodin/vesta#641]
This isn't so much a bug or feature request, but just a repository management type of thing.

You take a long time accepting PR's because of life, and this can be quite serious in the event of a serious issue in the code (XSS, etc).

I propose that you elect 5 or so contributors that come here a lot (see the "Contributors" thing, pick the ones you think would review/reply quicker), and ones that know the code base pretty well - just to get these PR's sorted quicker, and issues handled faster.

Just an idea, open to discussion.

_Side note: don't go electing me. I barely know where to find things on here... the code base is new to me._

**Comment by serghey-rodin** [comment]
Good idea, but why not you? I think you are good ad shell stuff :)
So would you like to take responsibility to check system related PRs ?

**Comment by SysVoid** [comment]
I've not worked on the shell stuff here. :P

I've only done the web stuff.

And if anything I'd only be able to assist with some basic things, or I'd have to get studying with the code lmao.

But if you do elect me, pick some others too. You really need a range of people so that all types of PR's and issues can be reviewed.

**Comment by EminezArtus** [comment]
Not fair since last year I contributed a ton of code...

**Comment by SysVoid** [comment]
@EminezArtus then you are in the group of people that I said should be added.

**Comment by SysVoid** [comment]
@EminezArtus just took a look. You're not even on the [Contributors Page](https://github.com/serghey-rodin/vesta/graphs/contributors).

Did you make PR's when you edited it?

**Comment by EminezArtus** [comment]
lol I contribute a ton of code that sat there for months so I removed all my code.

Only this remains:
https://github.com/serghey-rodin/vesta/issues/created_by/EminezArtus

**Comment by SysVoid** [comment]
Unless your code gets put in, most people wouldn't consider you as someone who should be managing PR's.

People like @very-twi should get RW access.

**Comment by EminezArtus** [comment]
To be honest I dont care much. I'm working on a new python CP. I would like to mention that I released a centos 7 version of vesta way before vesta did, because of all the time wasted on the file manager and other none important things.

**Comment by SysVoid** [comment]
@EminezArtus then why bother debating? :P

**Comment by EminezArtus** [comment]
lol im not i just said its not fair lol

**Comment by EminezArtus** [comment]
I was giving @serghey-rodin a hard time :D I think your post is good for VestaCP Project.

**Comment by KeiroD** [comment]
I'd be willing to help with approving PRs, checking code, that sort of thing. I do a fair amount of help as well.

Ironically, I've been working on getting some form of automated testing in place for VestaCP... but so far, that's been a giant pain in the ASS to get working... but I'm sure I'll figure it out at some point.

**Comment by SysVoid** [comment]
@KeiroD I have that too. :D

**Comment by tjebbeke** [comment]
I don't think Vesta need people who have contributed much code. I think Vesta needs some people who are active now. Who wants to check the code and fix the bugs, who makes time helping others on the forum. What I would do is creating a branch for every version, so they can fix critical bugs with a new release. Also, it's easier when pull requests are merged faster to add new features.

**Comment by KeiroD** [comment]
Personally, I don't mind offering my time in doing that. I'm hoping to get better testing for our code into place as soon as possible, but that's taking me some time to get working properly.

**Comment by tjebbeke** [comment]
We don't need to work fast. We need quality. It isn't good that five people are working individually on their own thing. We need to discuss all the tools and choose one and work all on that tool.

**Comment by KeiroD** [comment]
That's why I'm working on a solution for us to automate building so we can have an easier time in testing the code changes that we make.

I'll be working on getting the solution working. Once I have it working, I'll be submitting a PR to get the automated build system in place so we can more easily build VestaCP and look for bugs that way.

**Comment by tjebbeke** [comment]
That's why I said we need to discuss the tools together before starts working on it. Because what if someone, like the project owner, doesn't know how to work with your tool? Then you made it all for nothing. Or what if someone else is doing the same? Teamwork is important to reach our goal. So first we need to develop a strategy together, and then we can all do a little part and create a big thing together.

**Comment by n1trux** [comment]
**tl;dr:** make @SysVoid and @very-twi contributors and let them do stuff.

---

Commenting and subscribing to this issue. Project management for vesta is a mess, we have 2 separate bug trackers (one on bugs.vestacp.com and Github Issues).
I already proposed consolidating them, either closing Github issues because of the newbies who might post here (much higher barrier registering on a different bug tracker, so less questions and more actual issues by people who care about the project) or closing bugs.vestacp.com because that needs maintenance, too.

I don't think that a lot of posts in cyrillic (Moldovian?) are helping the cause – when I can't read the issues, I can't fix them. And I don't want to rely on Google translate. I feel that stuff like #532 should be addressed too since that'd make contributing and merging code way easier. You shouldn't push beta code in production.

@EminezArtus instead of being butthurt you should have rather addressed the concerns of the project owner and proposed a solution. You mentioned #285, which is a really huge PR – maybe split that in smaller chunks, like @serghey-rodin already suggested:

>Your Pull Request include many new cool things, but number of changes makes it harder to verify and review. If you are still willing to help us make Vesta better, could you please create separate PRs for each feature. I will review them as fast as I can. Thank you very much

:+1: for @SysVoid being elected as contributor since I feel he's capable of reading code and understands the FLOSS ecosystem. I wouldn't add too many contributors since too many cooks spoil the broth. I agree with SysVoid's proposal for giving @very-twi R/W access though since he seems to have contributed a lot of HQ stuff continuously.

**Comment by SysVoid** [comment]
@tjebbeke those are some good points. I agree that we should definitely pick some active members to just do basic maintenance.

**Comment by EminezArtus** [comment]
No my PRs were seperate i bundled everything as a last resort because this project was going no where. The owner of the project was to busy building paid products to address all the contributers thats why I ported the code and made my own version for centos 7. I came on here to support the post, because I know first hand the need for people. FWI there was no need for someone to test my code. It was already tested and if you look at my Big PR it was mostly essential fixes.

**Comment by KeiroD** [comment]
@EminezArtus, Yeah, those are pretty essential fixes... I don't know why your PR was rejected, as it should not have been.

We've had VestaCP set itself up with its self-signed certificates so I know that's not the issue.

@tjebbeke the automated build process that I've been working on setting up is mostly set, fire and forget. I've been working on integrating Travis CI into the PRs so we can have Travis CI check over the PRs and see if the code integrates nicely. If it does, it passes. If it doesn't, it fails. And it will link you to the build log.

If you'd like an example of such a thing, check out https://github.com/KeiroD/inspircd where I've automated the build process for stable and master there.

Simply put, I've been working on getting Travis CI to the point where we can just have it seamlessly integrated into everything. That's why I'm still working on it pretty heavily.

**Comment by Arinerron** [comment]
Since this is a democracy, I'm going to vote for @SysVoid. :wink:

**Comment by n1trux** [comment]
@Arinerron but SysVoid is already a collaborator!

We should instead focus on testing and reviewing the code base faster.
